cohttp/src/conn_cache.ml

```diff
@@ -0,0 +1,10 @@
+module type S = sig
```

Comment:
This signature should be in `cohttp-lwt`.

Could we call it `Connection`, as in https://github.com/mirage/ocaml-cohttp/blob/master/cohttp-async/src/client.mli#L30? I understand that here we are not exactly doing the same, but is there a way to make the interfaces similar?

Reply:
Ah, indeed. But I'd rather rename
- `Pipeline` → `Connection`
- `Pipeline.process` → `Connection.request`
- `Pipeline.shutdown` → `Connection.close`
- `Connection.is_closed` could be implemented by using the already present condition variable.
Last time I looked there was no connection cache in cohttp-async.

I'll do the renaming. But not tonight.

Reply:
Great!

Reply:
OK. Renaming is finished. But I'm reluctant to use the parameter ordering of `Cohttp_async.Client.Connection.request`:
```
val Cohttp_async.Client.Connection.request :
  ?body:Body.t ->
  t ->
  Http.Request.t ->
  (Http.Response.t * Body.t) Async_kernel.Deferred.t
```
But rather keep my ordering:
```
val Cohttp_lwt.Client.Connection.request :
  t ->
  ?body:Body.t ->
  Cohttp.Request.t ->
  (Cohttp.Response.t * Body.t) Lwt.t
```
since you might very well want to do a partial application of just `t` and be able to supply the body later. While there is no use in eliminating the `?body` optional parameter early.

Reply:
I don't mind keeping your ordering, I always thought that with labeled parameter it did not matter to be honest. If this is the case it may be worth discussing if we should swap the async call (ping @anuragsoni) as well

Reply:
Hmm, I also thought that with optional arguments it did not impact partial application in this case since we do have a `Request` that follows `t`.